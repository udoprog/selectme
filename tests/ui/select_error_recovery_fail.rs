pub(crate) fn main() {
}

/// Failed to parse the first block and recover to parse the second one (and
/// avoid emitting an error for it).
async fn error_recovery() {
    selectme::select! {
        binding => {}
    };

    selectme::select! {
        binding0 {} if
        binding1 = expr => {}
        binding2 {}
    };

    selectme::select! {
        binding0 = {} if
        binding1 = expr => {}
        binding2 = {}
    };

    selectme::select! {
        binding0 = expr0 {} if
        binding1 = expr1 => {}
        binding2 = expr2 {}
    };
}
