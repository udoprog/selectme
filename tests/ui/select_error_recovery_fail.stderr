error: binding must be followed by a `=`
 --> tests/ui/select_error_recovery_fail.rs:8:17
  |
8 |         binding => {}
  |                 ^

error: binding must be followed by a `=`
  --> tests/ui/select_error_recovery_fail.rs:12:21
   |
12 |         binding0 {} if
   |                     ^^

error: binding must be followed by a `=`
  --> tests/ui/select_error_recovery_fail.rs:14:18
   |
14 |         binding2 {}
   |                  ^^

error: expected `=>`
  --> tests/ui/select_error_recovery_fail.rs:19:18
   |
19 |         binding1 = expr => {}
   |                  ^

error: expected branch expression followed by `=>`
  --> tests/ui/select_error_recovery_fail.rs:20:20
   |
20 |         binding2 = {}
   |                    ^^

error: expected `=>`
  --> tests/ui/select_error_recovery_fail.rs:25:18
   |
25 |         binding1 = expr1 => {}
   |                  ^

error: expected branch expression followed by `=>`
  --> tests/ui/select_error_recovery_fail.rs:26:26
   |
26 |         binding2 = expr2 {}
   |                          ^^
